### PR TITLE
Add internal overrideable _flashFee in ERC20FlashMint

### LINKED ISSUE #3331: Add internal _flashFee without validation to ERC20FlashMint
**🧐 Motivation**

I have came across the below function where implementers can to override to customize the functionality.
The thing is the validation within the base implementation. Basically as a developer, I would expect the function to act as getter returning the fee amount while keeping all validations in place. But overriding this function leads to overriding the default validations as well.

```solidity
    function flashFee(address token, uint256 amount) public view virtual override returns (uint256) {
        require(token == address(this), "ERC20FlashMint: wrong token");
        // silence warning about unused variable without the addition of bytecode.
        amount;
        return 0;
    }
```

Similar case can be seen in the below function. But this time the base implementation has assumption than can be overloaded as well.
```solidity
    function maxFlashLoan(address token) public view virtual override returns (uint256) {
        return token == address(this) ? type(uint256).max - ERC20.totalSupply() : 0;
    }
```

I believe such flow (validation/assumption within override functions) is dangerous and should be addressed to mitigate the risks that could arise in implementation contracts.

### PR DESCRIPTION
**Pull request by nirban256**
Fixes #3331  

added internal _flashFee function 

added the function so that instead of ```flashFee``` function the ```_flashFee``` function can be overridden inside the ERC20FlashMint contract

#### PR Checklist

- [x] Tests
- [ ] Documentation

### CODE REVIEW

**Comment by Amxx** [comment]
Hello @nirban256 

I'm really puzzeled by this design.

```
    function flashFee(address token, uint256 amount) public view virtual override returns (uint256) {
        require(token == address(this), "ERC20FlashMint: wrong token");
        // silence warning about unused variable without the addition of bytecode.
        amount;
        return 0;
    }
    function _flashFee(address token, uint256 amount) internal view virtual returns (uint256) {
        uint256 fee = flashFee(token, amount);
        return fee;
    }
```

What are dev supposed to override? If you override the public one, you risk losing the `require`. If you override the internal, you risk having an inconsistency between the public value and the value used.

**Comment by Amxx** [comment]
Maybe what we want is

```
    function flashFee(address token, uint256 amount) public view virtual override returns (uint256) {
        require(token == address(this), "ERC20FlashMint: wrong token");
        return _flashFee(token, amount);
    }
    function _flashFee(address token, uint256 amount) internal view virtual returns (uint256) {
        // silence warning about unused variable without the addition of bytecode.
        token;
        amount;
        return 0;
    }
```

**Comment by nirban256** [comment]
Ok I will change this to this new implementation shortly

File: CHANGELOG.md [hunk 1]
```diff
@@ -10,6 +10,7 @@
  * `ERC721`: optimize transfers by making approval clearing implicit instead of emitting an event. ([#3481](https://github.com/OpenZeppelin/openzeppelin-contracts/pull/3481))
  * `ERC721`: optimize burn by making approval clearing implicit instead of emitting an event. ([#3538](https://github.com/OpenZeppelin/openzeppelin-contracts/pull/3538))
  * `ReentrancyGuard`: Reduce code size impact of the modifier by using internal functions. ([#3515](https://github.com/OpenZeppelin/openzeppelin-contracts/pull/3515))
+ * `ERC20FlashMint`: add `_flashFee` function for calling flashFee inside it. ([#3551](https://github.com/OpenZeppelin/openzeppelin-contracts/pull/3551))
```
Comment (frangio) [inline CHANGELOG.md:13]:
```suggestion
 * `ERC20FlashMint`: add an internal `_flashFee` function for overriding. ([#3551](https://github.com/OpenZeppelin/openzeppelin-contracts/pull/3551))
```

**Review by frangio — COMMENTED** [review]
Please update `ERC20FlashMintMock` to override `_flashFee` instead of `flashFee` like it does currently.

File: CHANGELOG.md (hunk 1 above)
Comment (nirban256) [reply CHANGELOG.md]: did I made the correct changes?

Comment (frangio) [reply CHANGELOG.md]: No, this changelog entry was describing the original contents of the PR.

**Comment by frangio** [comment]
My previous comment hasn't been addressed yet:
> Please update ERC20FlashMintMock to override _flashFee instead of flashFee like it does currently.
It refers to the mock contract used for testing.

**Comment by nirban256** [comment]
> 

Sir I have pushed a commit, let me know what are the changes required.

**Review by frangio — APPROVED** [review]
Thanks!

**Review by frangio — APPROVED** [review]

**Pull request merged**